tee-worker/litentry/pallets/identity-management/src/identity_context.rs

```diff
@@ -101,10 +103,26 @@ pub fn get_eligible_identities<T: Config>(
 				//   care about web2 identities, this step will empty `IdentityContext.web3networks`
 				// - it helps to reduce the request size a bit
 				networks.retain(|n| desired_web3networks.contains(n));
-				if networks.is_empty() && item.0.is_web3() {
-					None
+
+				if force_retain_web2_identity && item.0.is_web2() {
+					return Some((item.0.clone(), Vec::new()))
```

Comment:
This seems odd, Like we always know an `IdentityNetworkTuple` for a `web2` identity will always be `(Identity(//web2), vec![])` , is there a way we can improve this logic by changing/modifying the structs to support this usecase and any future usecases?

Reply:
I think it lies in the fact that we might use both web2 and web3 identities for an assertion building. I can imagine the best solution is have a `Vec<Network>` instead of `Vec<Web3Network>` to basically as web2 network types to the enum too.

This will be a breaking change tho, and only A1 will benefit from it (for now), so I didn't bother doing it 😂